pkg/host/internal/bridge/ovs/ovs.go

```diff
@@ -0,0 +1,677 @@
+	portEntryList := []*PortEntry{}
+	err := dbClient.WhereAll(portEntry, model.Condition{
+		Field:    &portEntry.Interfaces,
+		Function: ovsdb.ConditionIncludes,
+		Value:    []string{iface.UUID},
+	}).List(ctx, &portEntryList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list ports related to interface %s: %v", iface.Name, err)
+	}
+	if len(portEntryList) == 0 {
+		return nil, nil
+	}
+	return portEntryList[0], nil
+}
+
+func (o *ovs) getBridgeByPort(ctx context.Context, dbClient client.Client, port *PortEntry) (*BridgeEntry, error) {
+	brEntry := &BridgeEntry{}
+	brEntryList := []*BridgeEntry{}
+	err := dbClient.WhereAll(brEntry, model.Condition{
+		Field:    &brEntry.Ports,
+		Function: ovsdb.ConditionIncludes,
+		Value:    []string{port.UUID},
+	}).List(ctx, &brEntryList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list bridges related to port %s: %v", port.Name, err)
+	}
+	if len(brEntryList) == 0 {
+		return nil, nil
+	}
+	return brEntryList[0], nil
+}
+
+// create bridge with provided configuration
+func (o *ovs) createBridge(ctx context.Context, dbClient client.Client, br *BridgeEntry) error {
+	brCreateOps, err := dbClient.Create(br)
+	if err != nil {
+		return fmt.Errorf("failed to prepare operation for bridge creation: %v", err)
+	}
+	rootObj, err := o.getRootObj(ctx, dbClient)
+	if err != nil {
+		return err
+	}
+	ovsMutateOps, err := dbClient.Where(rootObj).Mutate(rootObj, model.Mutation{
+		Field:   &rootObj.Bridges,
+		Mutator: ovsdb.MutateOperationInsert,
+		Value:   []string{br.UUID},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create mutate operation for Open_vSwitch table: %v", err)
+	}
+	if err := o.execTransaction(ctx, dbClient, brCreateOps, ovsMutateOps); err != nil {
+		return fmt.Errorf("bridge creation failed: %v", err)
+	}
+	return nil
+}
+
+// add interface with provided configuration to the provided bridge
+// and check that interface has no error for the next 2 seconds
+func (o *ovs) addInterface(ctx context.Context, dbClient client.Client, br *BridgeEntry, iface *InterfaceEntry) error {
+	addInterfaceOPs, err := dbClient.Create(iface)
+	if err != nil {
+		return fmt.Errorf("failed to prepare operation for interface creation: %v", err)
+	}
+	port := &PortEntry{Name: iface.Name, UUID: uuid.NewString(), Interfaces: []string{iface.UUID}}
+	addPortOPs, err := dbClient.Create(port)
+	if err != nil {
+		return fmt.Errorf("failed to prepare operation for port creation: %v", err)
+	}
+	bridgeMutateOps, err := dbClient.Where(br).Mutate(br, model.Mutation{
+		Field:   &br.Ports,
+		Mutator: ovsdb.MutateOperationInsert,
+		Value:   []string{port.UUID},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to prepare operation for bridge mutate: %v", err)
+	}
+	if err := o.execTransaction(ctx, dbClient, addInterfaceOPs, addPortOPs, bridgeMutateOps); err != nil {
+		return fmt.Errorf("bridge deletion failed: %v", err)
+	}
+	// check after ~2 seconds that interface has no error
```

Comment:
You mean timeout or enable/disable of the check?
For both cases, I would prefer to keep it as is for now. The main goal of this check is to try to avoid situations when we silently proceed with the configuration even if the interface is not connected to the switch properly (Interface create call will not return an error from ovsdb in this case). The only way to check that the interface attachement failed is to check the error field for the interface after some time after creation.

Reply:
ack, yea i also dont want it configurable :). i think what i meant is : can you make this one a const/var like OVSIntefaceWaitTimeoutSeconds or similar ?

Reply:
ack. makes sense, done